server/region_syncer/server.go

```diff
@@ -89,28 +94,86 @@ func (s *RegionSyncer) RunServer(regionNotifier <-chan *core.RegionInfo, quit ch
 			return
 		case first := <-regionNotifier:
 			requests = append(requests, first.GetMeta())
-			s.history.record(first)
+			startIndex := s.history.GetNextIndex()
+			s.history.Record(first)
 			pending := len(regionNotifier)
 			for i := 0; i < pending && i < maxSyncRegionBatchSize; i++ {
 				region := <-regionNotifier
 				requests = append(requests, region.GetMeta())
-				s.history.record(region)
+				s.history.Record(region)
 			}
 			regions := &pdpb.SyncRegionResponse{
-				Header:  &pdpb.ResponseHeader{ClusterId: s.server.ClusterID()},
-				Regions: requests,
+				Header:     &pdpb.ResponseHeader{ClusterId: s.server.ClusterID()},
+				Regions:    requests,
+				StartIndex: startIndex,
 			}
 			s.broadcast(regions)
 		case <-ticker.C:
-			alive := &pdpb.SyncRegionResponse{Header: &pdpb.ResponseHeader{ClusterId: s.server.ClusterID()}}
+			alive := &pdpb.SyncRegionResponse{
+				Header: &pdpb.ResponseHeader{ClusterId: s.server.ClusterID()},
+				// use max uint64 to mark keepalive.
+				StartIndex: math.MaxUint64,
```

Comment:
I think it's ok to send `currentIndex` here. Do not need a special maxUint64 index.